test/query_test.rb

```diff
@@ -267,6 +274,13 @@ def test_difference
     assert_equal [@ref_n1], get_set_contents(set)
   end
 
+  def test_distinct
+    set = Query.expr { match(Query.ref('indexes/n_of_widgets')) }
+    assert_equal [0, 1, 1], get_set_contents(set)
```

Comment:
Tests are anyway independent of each other, right? Since `tearDown` deletes the DB, I think we should be ok in any case.